platform-sdk/swirlds-base/src/main/java/com/swirlds/base/utility/ToStringBuilder.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.base.utility;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A builder for toString methods.
+ * <p>
+ * It can be used to build a toString method for an object by appending values to the builder and then calling
+ * {@link #toString()} to get the final string. This maintains a uniform toString format across the codebase.
+ * <p>
+ * If you are using IntelliJ, you can use the "Generate" menu to generate a toString method using this class by adding
+ * the following templates:
+ * <pre>
+ * public java.lang.String toString() {
```

Comment:
we need to add the documentation to https://github.com/hashgraph/hedera-services/tree/develop/documentation